src/GUI/RevEng.Shared/TableInformationModel.cs

```diff
@@ -47,20 +48,37 @@ public bool HasPrimaryKey
             }
         }
 
+        /// <summary>
+        /// Gets or sets whether the table or view is keyless - always false for EF Core 2.0.
+        /// </summary>
+        [IgnoreDataMember]
+        public bool IsKeyless
+        {
+            get => _isKeyless;
+            set
+            {
+                if (value == _isKeyless) return;
+                _isKeyless = value;
+                OnPropertyChanged();
+            }
+        }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="TableInformationModel"/> class for a specific table.
         /// </summary>
         /// <param name="name">The table name.</param>
         /// <param name="hasPrimaryKey">Whether or not a primary key exists for the table.</param>
         /// <exception cref="ArgumentException"><paramref name="schema"/> or <paramref name="name"/> are null or only white spaces.</exception>
         public TableInformationModel(string name,
-                                     bool hasPrimaryKey)
+                                     bool hasPrimaryKey,
+                                     bool isKeyless = false)
         {
             if (string.IsNullOrWhiteSpace(name))
                 throw new ArgumentException(@"Value cannot be empty or only white spaces.", nameof(name));
 
```

Comment:
Is every combination of `HasPrimaryKey` and `IsKeyless` valid? If not, throw an exception in the constructor?